fdbclient/NativeAPI.actor.cpp

```diff
@@ -3389,16 +3450,33 @@ SpanContext generateSpanID(bool transactionTracingSample, SpanContext parentCont
 
 ACTOR Future<int64_t> lookupTenantImpl(DatabaseContext* cx, TenantName tenant) {
 	loop {
-		++cx->transactionTenantLookupRequests;
-		choose {
-			when(wait(cx->onProxiesChanged())) {}
-			when(GetTenantIdReply rep = wait(basicLoadBalance(cx->getCommitProxies(UseProvisionalProxies::False),
-			                                                  &CommitProxyInterface::getTenantId,
-			                                                  GetTenantIdRequest(tenant, latestVersion),
-			                                                  TaskPriority::DefaultPromiseEndpoint))) {
-				++cx->transactionTenantLookupRequestsCompleted;
-				return rep.tenantId;
+		try {
+			double backoff = cx->getBackoff();
+			if (backoff > 0.0) {
+				wait(delay(backoff));
+			}
```

Comment:
Since all of the callers of this are getting the delay and then conditionally waiting, you could potentially write `getBackoff` to return a future like:

```
return backoff > 0.0 ? delay(backoff) : Future<Void>(Void());
```

and then these callers could get simpler:

```suggestion
			wait(cx->getBackoff());
```

I'll leave that up to you, though.

Reply:
Good idea! Change applied.